code/game/objects/items/toys.dm

```diff
@@ -1086,3 +1086,63 @@
 	desc = "Small mechanical canary in a cage, does absolutely nothing of any importance!"
 	icon = 'icons/mob/animal.dmi'
 	icon_state = "canary"
+
+/obj/item/toy/ball
+	name = "ball"
+	desc = "A ball!"
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "beach_ball"
+	w_class = W_CLASS_TINY //So they go further when kicking
+
+/obj/item/toy/ball/afterattack(atom/target as mob|obj|turf|area, mob/user as mob)
+	if(user.drop_item(src))
+		src.throw_at(target, throw_range, throw_speed)
+
+/obj/item/toy/ball/throw_at(atom/target, range, speed, override = 1) //Shamelessly stolen from boomerang.dm
+	if(!usr)
+		return ..()
+
+	spawn()
+		animate(src, transform = turn(matrix(), 120), time = 5, loop = -1)
+		animate(transform = turn(matrix(), 240), time = 5)
+		animate(transform = null, time = 5)
+
+		while(throwing)
+			sleep(5)
+
+		animate(src) //Stop the animation
+
+	..()
+
+/obj/item/toy/ball/throw_impact(atom/hit_atom, var/speed, user)
+	if(iscarbon(hit_atom) && !isslime(hit_atom))
+		var/mob/living/carbon/L = hit_atom
+		if(L.get_active_hand() == null)
```

Comment:
No need for the L.get_active_hand() == null when you can use !L.get_active_hand()